pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -208,6 +208,9 @@ func RunStartBuild(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra
 			},
 			Commit: commit,
 		}
+		if len(env) > 0 {
+			fmt.Fprintf(cmd.Out(), "WARNING: The --env and --build-loglevel flags are not supported if the binary source is used.\n")
```

Comment:
@smarterclayton well users don't know that --build-loglevel is setting an env behind them ;-)